tests/test_state_estimator.py

```diff
@@ -32,33 +32,87 @@ class TestStateEstimator(unittest.TestCase):
     def setUp(self):
         self.ofc_data = OFCData("lsst")
 
-        self.estimator = StateEstimator(self.ofc_data)
+        self.estimator = StateEstimator(self.ofc_data, rcond=5e-4)
 
         self.wfe = np.loadtxt(
             pathlib.Path(__file__).parent.absolute()
             / "testData"
             / "lsst_wfs_error_iter0.z4c"
         )
 
-        sensor_name_list = ["R44_SW0", "R04_SW0", "R00_SW0", "R40_SW0"]
+        dofs = np.loadtxt(
+            pathlib.Path(__file__).parent.absolute() / "testData" / "lsst_dof_iter0.txt"
+        )
+        self.dofs = dofs[:, 1]
+
+        # Constuct the double zernike sensitivity matrix
+        self.dz_sensitivity_matrix = SensitivityMatrix(self.ofc_data)
 
-        self.field_idx = [
-            self.estimator.ofc_data.field_idx[sensor_name]
-            for sensor_name in sensor_name_list
+        self.sensor_name_list = ["R00_SW0", "R04_SW0", "R40_SW0", "R44_SW0"]
+
+        self.field_angles = [
+            self.ofc_data.sample_points[sensor] for sensor in self.sensor_name_list
         ]
 
+    def mean_squared_residual(self, new_array, reference_array):
+        return np.sum((new_array - reference_array) ** 2) / np.sum(reference_array**2)
+
+    def compute_sensitivity_matrix(self, field_angles, rotation_angle):
+        # Evaluate sensitivity matrix at sensor positions
+        sensitivity_matrix = self.dz_sensitivity_matrix.evaluate(
+            field_angles, rotation_angle
+        )
+
+        # Select sensitivity matrix only at used degrees of freedom
+        sensitivity_matrix = sensitivity_matrix[
+            :, self.dz_sensitivity_matrix.ofc_data.zn3_idx, :
+        ]
+
+        # Reshape sensitivity matrix to dimensions
+        # (#zk * #sensors, # dofs) = (19 * #sensors, 50)
+        size_ = sensitivity_matrix.shape[2]
+        sensitivity_matrix = sensitivity_matrix.reshape((-1, size_))
```

Comment:
Please avoid to have the variable names that have the `_` at the end.
Is there any reason that you could not name the `size` but `size_`?

Reply:
Done! No particular reason. Fixed it.